### ISSUE TITLE
[no-unused-modules] `import type` doesn't mark export as used.

### ISSUE THREAD

**Issue by phaux** [benmosher/eslint-plugin-import#1950]
Repro:

```ts
// Foo.ts
export interface Foo {}

// index.ts
import type { Foo } from "./Foo"
```

> exported declaration 'Foo' not used within other modules

Config:

```json5
{
  "extends": [
    "plugin:import/recommended",
    "plugin:import/typescript",
    // and more ...
  ],
  "rules": {
    "import/no-unused-modules": ["error", { "unusedExports": true }],
  },
}
```

Versions:

```
@typescript-eslint/parser: 4.8.0
eslint: 7.13.0
eslint-plugin-import: 2.22.1
```

**Comment by mironiasty** [comment]
I think it is same issue as reported here: https://github.com/benmosher/eslint-plugin-import/issues/1924

**Comment by ljharb** [comment]
Duplicate of #1924. Fixed by #1974.
